music21/tinyNotation.py

```diff
@@ -1256,6 +1375,90 @@ def testOne(self):
         self.assertEqual(sfn[12].duration.quarterLength, 1.0)
         self.assertEqual(sfn[12].expressions[0].classes, expressions.Fermata().classes)
 
+    def testRaiseExceptions(self) -> None:
+        error_states = [
+            {
+                "string": "h",
+                "reason": "h is not a valid note",
+            },
+            {
+                "string": "a;",
+                "reason": "a semicolon is not a valid character or modifier",
+            },
+            {
+                "string": "r;",
+                "reason": "a semicolon is not a valid character or modifier",
+            },
+            {
+                "string": "4/4;",
+                "reason": "a semicolon is not a valid character or modifier",
+            },
+            {
+                "string": "ABC",
+                "reason": (
+                    "only the same upper-cased letter may be repeated to "
+                    "indicate lower octaves"
+                ),
+            },
+        ]
+
+        for error_state in error_states:
+            with self.assertRaises(TinyNotationException, msg=(
+                    "Should have raised a TinyNotationException for input "
+                    f'"{error_state["string"]}" because {error_state["reason"]}.'
```

Comment:
Prefer single quotes for all code unless they also include single quotes (as in some tiny notation).

Please do not use implicit string concatenation.  It's a little faster than explicit concat, but tends to lead to bugs later.